kale/evaluate/uncertainty_metrics.py

```diff
@@ -0,0 +1,1036 @@
+import logging
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+
+from kale.prepdata.tabular_transform import apply_confidence_inversion
+
+
+def evaluate_bounds(
+    estimated_bounds: Dict[str, pd.DataFrame],
+    bin_predictions: Dict[str, pd.DataFrame],
+    uncertainty_pairs: List,
+    num_bins: int,
+    landmarks: List[int],
+    num_folds: int = 8,
+    show_fig: bool = False,
+    combine_middle_bins: bool = False,
+) -> Dict:
+    """
+    Evaluates error bounds for given uncertainty pairs and estimated bounds.
+
+    Args:
+        estimated_bounds (Dict[str, pd.DataFrame]): Dictionary of error bounds for each model.
+        bin_predictions (Dict[str, pd.DataFrame]): Dictionary of bin predictions for each model.
+        uncertainty_pairs (List[List[str]]): List of uncertainty pairs to be evaluated.
+        num_bins (int): Number of bins to be used.
+        landmarks (List[str]): List of landmarks to be evaluated.
+        num_folds (int, optional): Number of folds for cross-validation. Defaults to 8.
+        show_fig (bool, optional): Flag to show the figure. Defaults to False.
+        combine_middle_bins (bool, optional): Flag to combine the middle bins. Defaults to False.
+
+    Returns:
+        Dict: Dictionary containing evaluation results.
+    """
+
+    if combine_middle_bins:
+        num_bins = 3
+
+    # Initialise results dicts
+    all_bound_percents = {}
+    all_bound_percents_nolmsep = {}
+
+    all_concat_errorbound_bins_lm_sep_foldwise = [{} for x in range(len(landmarks))]  # type: List[Dict]
+    all_concat_errorbound_bins_lm_sep_all = [{} for x in range(len(landmarks))]  # type: List[Dict]
```

Comment:
what is lm in the variable name?

Reply:
lm = LandMark.

Improvement: all_concat_errorbound_bins_target_sep_foldwise